core/object.go

```diff
@@ -1003,6 +1003,75 @@ func (bi *BigInt) Compare(other Object) int {
 	return CompareNumbers(bi, EnsureObjectIsNumber(other, "Cannot compare BigInt: %s"))
 }
 
+// Determine the precision for a float-point constant, with float64
+// precision (53) being the minimum value. No need to be as strict
+// when parsing it as is math/big.(Float).Parse().
+func computePrecision(s string) (prec uint) {
+	prec = 53 // Default to precision for float64
+	if s == "" {
+		return
+	}
+	if s[0] == '-' || s[0] == '+' {
+		s = s[1:]
+	}
+	if s == "Inf" || s == "inf" || s == "NaN" {
+		return
+	}
+
+	bitsNeeded := 0.
+
+	// Assume base 10 at first.
+	bitsPerDigit := 3.3 // (joker.math/log-2 10) => 3.32192809488736
```

Comment:
Should not it be more precise value (and rounded up) like 3.322 or otherwise it may not be enough to represent all digits?

Reply:
Huh, I just plain rounded up incorrectly. `3.33` seems better. Fixed!